unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,258 @@
+ *                      -# Calling the ilm_init()
+ *                      -# Calling the ilm_isInitialized()
+ *                      -# Verification point:
+ *                         +# ilm_init() must return ILM_SUCCESS
+ *                         +# ilm_isInitialized() must return ILM_TRUE
+ */
+TEST_F(IlmCommonTest, ilm_isInitialized_getTrue)
+{
+    mock_ilmInitSuccess();
+
+    ASSERT_EQ(ILM_SUCCESS, ilm_init());
+    ASSERT_EQ(ILM_TRUE, ilm_isInitialized());
+}
+
+/** ================================================================================================
+ * @test_id             ilm_registerShutdownNotification_getFailure
+ * @brief               Test case of ilm_registerShutdownNotification()
+ *                      where the internal function ilmControl_registerShutdownNotification fails, return ILM_FAILED
+ * @test_procedure Steps:
+ *                      -# Mocking the ilmControl_registerShutdownNotification() does return ILM_FAILED
+ *                      -# Calling the ilm_registerShutdownNotification()
+ *                      -# Verification point:
+ *                         +# ilm_registerShutdownNotification() must return ILM_FAILED
+ *                         +# ilmControl_registerShutdownNotification() must be called once time and retrun ILM_FAILED
+ */
+TEST_F(IlmCommonTest, ilm_registerShutdownNotification_getFailure)
+{
+    mp_ilmErrorType[0] = ILM_FAILED;
+    SET_RETURN_SEQ(ilmControl_registerShutdownNotification, mp_ilmErrorType, MAX_NUMBER);
+
+    ASSERT_EQ(ILM_FAILED, ilm_registerShutdownNotification(nullptr, nullptr));
+
+    ASSERT_EQ(1, ilmControl_registerShutdownNotification_fake.call_count);
+    ASSERT_EQ(mp_ilmErrorType[0], ilmControl_registerShutdownNotification_fake.return_val_history[0]);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_registerShutdownNotification_getSuccess
+ * @brief               Test case of ilm_registerShutdownNotification()
+ *                      where the internal function ilmControl_registerShutdownNotification() success, return ILM_SUCCESS
+ * @test_procedure Steps:
+ *                      -# Mocking the ilmControl_registerShutdownNotification() does return ILM_SUCCESS
+ *                      -# Calling the ilm_registerShutdownNotification()
+ *                      -# Verification point:
+ *                         +# ilm_registerShutdownNotification() must return ILM_SUCCESS
+ *                         +# ilmControl_registerShutdownNotification() must be called once time and retrun ILM_SUCCESS
+ */
+TEST_F(IlmCommonTest, ilm_registerShutdownNotification_getSuccess)
+{
+    mp_ilmErrorType[0] = ILM_SUCCESS;
+    SET_RETURN_SEQ(ilmControl_registerShutdownNotification, mp_ilmErrorType, MAX_NUMBER);
+
+    ASSERT_EQ(ILM_SUCCESS, ilm_registerShutdownNotification(nullptr, nullptr));
```

Comment:
In cases where api makes a call to sync_and_acquire_instance we should check for call_count matches the release_instance call_count.